test/test_query.py

```diff
@@ -714,6 +714,206 @@ def test_match_after_set_none(self):
+                                       dbcore.query.SubstringQuery)
+        not_results = self.lib.items(dbcore.query.NotQuery([q]))
+        self.assert_items_matched(not_results, ['baz qux'])
+        self.assertNegationProperties(q)
+
+    def test_type_boolean(self):
+        q = dbcore.query.BooleanQuery('comp', True)
+        not_results = self.lib.items(dbcore.query.NotQuery([q]))
+        self.assert_items_matched(not_results, ['beets 4 eva'])
+        self.assertNegationProperties(q)
+
+    def test_type_date(self):
+        q = dbcore.query.DateQuery('mtime', '0.0')
+        not_results = self.lib.items(dbcore.query.NotQuery([q]))
+        self.assert_items_matched(not_results, [])
+        self.assertNegationProperties(q)
+
+    def test_type_false(self):
+        q = dbcore.query.FalseQuery()
+        not_results = self.lib.items(dbcore.query.NotQuery([q]))
+        self.assert_items_matched_all(not_results)
+        self.assertNegationProperties(q)
+
+    def test_type_match(self):
+        q = dbcore.query.MatchQuery('year', '2003')
+        not_results = self.lib.items(dbcore.query.NotQuery([q]))
+        self.assert_items_matched(not_results, ['foo bar', 'baz qux'])
+        self.assertNegationProperties(q)
+
+    def test_type_none(self):
+        q = dbcore.query.NoneQuery('rg_track_gain')
+        not_results = self.lib.items(dbcore.query.NotQuery([q]))
+        self.assert_items_matched(not_results, [])
+        self.assertNegationProperties(q)
+
+    def test_type_numeric(self):
+        q = dbcore.query.NumericQuery('year', '2001..2002')
+        not_results = self.lib.items(dbcore.query.NotQuery([q]))
+        self.assert_items_matched(not_results, ['beets 4 eva'])
+        self.assertNegationProperties(q)
+
+    def test_type_or(self):
+        # not(a or b) <-> not(a) and not(b)
+        q = dbcore.query.OrQuery([dbcore.query.BooleanQuery('comp', True),
+                                  dbcore.query.NumericQuery('year', '2002')])
+        not_results = self.lib.items(dbcore.query.NotQuery([q]))
+        self.assert_items_matched(not_results, ['beets 4 eva'])
+        self.assertNegationProperties(q)
+
+    def test_type_regexp(self):
+        q = dbcore.query.RegexpQuery('artist', '^t')
+        not_results = self.lib.items(dbcore.query.NotQuery([q]))
+        self.assert_items_matched(not_results, ['foo bar'])
+        self.assertNegationProperties(q)
+
+    def test_type_substring(self):
+        q = dbcore.query.SubstringQuery('album', 'ba')
+        not_results = self.lib.items(dbcore.query.NotQuery([q]))
+        self.assert_items_matched(not_results, ['beets 4 eva'])
+        self.assertNegationProperties(q)
+
+    def test_type_true(self):
+        q = dbcore.query.TrueQuery()
+        not_results = self.lib.items(dbcore.query.NotQuery([q]))
+        self.assert_items_matched(not_results, [])
+        self.assertNegationProperties(q)
+
+    def test_fast_vs_slow(self):
```

Comment:
This particular test is a bit ugly, but the idea is to make sure that both the `col_clause` and the `match` method are handled correctly.